### ISSUE TITLE
[RRFC] an ability for a package to indicate that it should always be installed as if --save-exact were specified

### ISSUE THREAD

**Issue by thorn0** [npm/rfcs#349]
## Motivation ("The Why")

> Install an exact version of Prettier locally in your project. This makes sure that everyone in the project gets the exact same version of Prettier. Even a patch release of Prettier can result in slightly different formatting, so you wouldn’t want different team members using different versions and formatting each other’s changes back and forth.

-- from Prettier's docs, https://prettier.io/docs/en/install.html#summary

### Example

Another example of where this can be really useful is packages, whose authors explicitly acknowledge they don't follow semver.
For example, TypeScript. See https://github.com/microsoft/TypeScript/issues/14116

#### Current Behaviour

There is no way to do this.

#### Desired Behaviour

As a package maintainer, I should be able to specify `"exactVersion": true` (or `"semver": false`) in `package.json`, so that the users automatically get `"my-package": "1.2.3"` instead of  `"my-package": "^1.2.3"` when they run `npm install my-package`.

**Comment by ljharb** [comment]
Saving an exact version isn’t the solution here, a lockfile is.

**Comment by thorn0** [comment]
Lockfiles are used (or not used) on the consumer's side. The one who installs the package makes that decision, not the one who publishes. Whereas I'm proposing to give the publisher a bit more control over this.

**Comment by ljharb** [comment]
Right - but it's supposed to be the consumer's decision. I would never, ever want to save prettier as an exact version, and i would actively want formatting changes to suddenly appear whenever i've installed something new on a package that lacks a lockfile.

Any project that cares about determinism has a lockfile already, otherwise they can't achieve that goal anyways.

**Comment by thorn0** [comment]
How does the default behavior of `npm install <package name without version>` fit into this line of reasoning? Shouldn't it add `"my-package": "*"` to `package.json` then instead of `"my-package": "^1.2.3"?

**Comment by thorn0** [comment]
If you have a lockfile, why be afraid of `*`?

**Comment by ljharb** [comment]
Because i don’t have a lockfile in any published packages - i only use a lockfile in apps. In an app, that would be quite safe - in a package, quite dangerous.

**Comment by thorn0** [comment]
In that case, for packages (not apps), using `^` for dependencies that don't adhere to semver is as bad as `*` is for those that do, isn't it? Why can't a package have an extra way to mitigate this by saying "hey! `npm install`'s default behavior doesn't make sense for me"?

**Comment by ljharb** [comment]
I agree it would make sense for a package author to be able to communicate that they don’t follow semver - typescript, for example - but prettier does follow semver to my understanding.
